### DISCUSSION TITLE
chia blockchain explorer project

### DISCUSSION THREAD

**Discussion by hajes** [Chia-Network/chia-blockchain#16987] [Support]
Is there a way to access local node sqlite3 database, please?

Such as https://xchscan.com/

I can access my transactions via RPC.

In latest Bkex exchange fiasco, authorities ask me for transactions to Bkex. I have address, but via public explorers there is no export function.

Python has got `sqlite3` module, can I simply access `.chia/mainnet/db/blockchain_v2_mainnet.sqlite`, please?

Filter according to address via SQL commands.

Any suggestions, please?

**Comment by BrandtH22** [comment]
Hey @hajes , as a heads up we can generally provide more timely and thorough support in our discord server (https://discord.gg/chia).

You can use the full node rpc  commands to pull data from that db:
https://docs.chia.net/full-node-rpc#get_coin_records_by_puzzle_hash

Note to go from the xch address to the puzzle hash you can use online tools like this one: 
https://www.spacescan.io/tools/converter/address

**Reply by hajes** [reply]
thanks for reply @BrandtH22, I am really blind...I was exploring DataLayer RPC...missed FullNode sections.

I have no Discord, and I do not like it. It is chaos where you cannot find anything on one place. Everything easily accessible on GitHub...anyone can find it ;-) The Keybase was same disaster. We old-school guys still prefer simplicity to whining in online chat full of clueless experts :-P

**Comment by hajes** [comment]
I have managed to pull all records associated with an address/puzzle hash

I see no info whether amounts were received/sent, and who sent/received it.

`        {
            "coin": {
                "amount": 4500000,
                "parent_coin_info": "...",
                "puzzle_hash": "..."
            },
            "coinbase": false,
            "confirmed_block_index": 3169408,
            "spent": true,
            "spent_block_index": 3202827,
            "timestamp": 1674898701
        },`

do I need to pull data from `parent_coin_info`, please?

**Reply by BrandtH22** [reply]
Hey @hajes , the coins puzzle_hash would be indicative of the "sender" when converted to the xch address. This can be done with the cdv tools (https://docs.chia.net/guides/crash-course/smart-coins/#creating-the-coin) or with tools provided by explorers (https://www.spacescan.io/tools/converter/address) . 

To determine the "recipient" one would need to get the child coin and perform a similar encoding of its puzzle hash.
You can use this RPC call with the previous calls coinID to find its children:
https://docs.chia.net/full-node-rpc#get_coin_records_by_parent_ids

Keep in mind that coins generally have 1 or more children, one will be the child coin which is now controlled by another user (i.e. had been sent to that other user) another will be the change coin (generally returned to the "sender") but there can be multiple other coins as well based on what type of spend it was

**Comment by hajes** [comment]
thanks chief
